### PR TITLE
Dev oct2015

### PR DESCRIPTION
**Pull request by gschmidt-usgs**
Modified convert_ozone.c, updatencep.py, updatetoms.py, and updatelads.py after the original peer review for this release.  Please take a look at these changes.  The main focus was to retry the wget calls in the event of an error.

### CODE REVIEW

File: not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py [hunk 1]
```diff
@@ -203,7 +208,20 @@ def downloadLads (year, doy, destination):
...
-        subprocess.call(cmd, shell=True, cwd=destination)
+        retval = subprocess.call(cmd, shell=True, cwd=destination)
+
+        # make sure the wget was successful or retry up to 5 more times and
+        # sleep in between
+        if retval:
+            retry_count = 1
+            while ((retry_count <= 5) and (retval)):
+                time.sleep(60)
+                print "Retry %d of wget for %s" % (retry_count, url)
```
Comment (dhill-usgs) [inline not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py]: calls to print should be a function now.  I think print statements are deprecated.

File: not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py [hunk 2]
```diff
@@ -203,7 +208,20 @@ def downloadLads (year, doy, destination):
...
+                subprocess.call(cmd, shell=True, cwd=destination)
```
Comment (dhill-usgs) [inline not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py]: does this ever get checked to see if it succeeds/fails?

File: not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py (hunk 1 above)
Comment (gschmidt-usgs) [inline not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py]: I need to do this globally in all my Python scripts.

File: not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py (hunk 2 above)
Comment (gschmidt-usgs) [inline not-validated-prototype-l8_sr/landsat_aux/scripts/updatelads.py]: Good catch and fixed.

File: ledaps/ledapsAncSrc/updatencep.py [hunk 3]
```diff
@@ -252,27 +257,40 @@ def cleanNcepTargetDir (ancdir, year):
...
-        print "unsuccessful download of %s" % url
-        return ERROR
+    # make sure the wget was successful or retry up to 5 more times and sleep
+    # in between
+    if retval:
+        retry_count = 1
+        while ((retry_count <= 5) and (retval)):
+            time.sleep(60)
+            print "Retry %d of wget for %s" % (retry_count, url)
+            subprocess.call(cmd, shell=True, cwd=destination)
```
Comment (dhill-usgs) [inline ledaps/ledapsAncSrc/updatencep.py]: assign ```retval``` here

Comment (gschmidt-usgs) [inline ledaps/ledapsAncSrc/updatencep.py]: Done and checked in with the fix for updatelads, updatencep, and updatetoms.

--------------------------------------------------------------
Gail Schmidt
Software Engineer
Stinger Ghaffarian Technologies (SGT, Inc.)
Contractor to the USGS Earth Resources Observation and Science (EROS) Center
Rapid City, SD  57701
605-593-2722   Phone
605-394-6078   Fax
gschmidt@usgs.gov

Adaptation - "Manage the unavoidable"
Mitigation - "Avoid the unmanageable"

**Comment by dhill-usgs** [comment]
i think the retval assignment needs to be checked for all the new inner loop calls to subprocess.  Otherwise nothing functional noticed.

A couple style notes, as previously said... 
* print statements are out and print() functions are in.
* python prefers snake_case for functions but there were are few camelCased()
* If some point in the future any rewriting happens, it might be worthwhile to replace the wget calls with pure python libs like requests, urllib2, or ftplib.  Doing this maintains the full stack tracing when an error occurs. (this is lost between the shell calls and python since we get stderr and retvals, not exceptions)

**Comment by gschmidt-usgs** [comment]
The retval is checked by the while loop around the retries.  If the
download is unsuccessful, the routine will not error out as a whole because
we need to support the case where the data isn't available and has a
latency for showing up on the ftp site.

--------------------------------------------------------------
Gail Schmidt
Software Engineer
Stinger Ghaffarian Technologies (SGT, Inc.)
Contractor to the USGS Earth Resources Observation and Science (EROS) Center
Rapid City, SD  57701
605-593-2722   Phone
605-394-6078   Fax
gschmidt@usgs.gov

Adaptation - "Manage the unavoidable"
Mitigation - "Avoid the unmanageable"

**Pull request merged**